extensions/wikia/Search/classes/Hooks.php

```diff
@@ -40,7 +40,11 @@ class Hooks {
 	 * @return true
 	 */
 	public static function onArticleDeleteComplete( &$article, \User &$user, $reason, $id ) {
-		return ( new Indexer )->deleteArticle( $id );
+		if ( $this->canIndex($article->getTitle()) ) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Remove this use of "$this". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS2014)